The dataloader test is failing on python 2.7 but looks unrelated.

<details>

```
Running dataloader tests
..............s....s....s....s.s.Traceback (most recent call last):
  File "test_dataloader.py", line 496, in <module>
    run_tests()
  File "/home/travis/build/pytorch/pytorch/test/common.py", line 35, in run_tests
    unittest.main(argv=UNITTEST_ARGS)
  File "/opt/python/2.7.14/lib/python2.7/unittest/main.py", line 95, in __init__
    self.runTests()
  File "/opt/python/2.7.14/lib/python2.7/unittest/main.py", line 232, in runTests
    self.result = testRunner.run(self.test)
  File "/opt/python/2.7.14/lib/python2.7/unittest/runner.py", line 151, in run
    test(result)
  File "/opt/python/2.7.14/lib/python2.7/unittest/suite.py", line 70, in __call__
    return self.run(*args, **kwds)
  File "/opt/python/2.7.14/lib/python2.7/unittest/suite.py", line 108, in run
    test(result)
  File "/opt/python/2.7.14/lib/python2.7/unittest/suite.py", line 70, in __call__
    return self.run(*args, **kwds)
  File "/opt/python/2.7.14/lib/python2.7/unittest/suite.py", line 108, in run
    test(result)
  File "/opt/python/2.7.14/lib/python2.7/unittest/case.py", line 393, in __call__
    return self.run(*args, **kwds)
  File "/opt/python/2.7.14/lib/python2.7/unittest/case.py", line 368, in run
    result.addSuccess(self)
    self.stream.write('.')
IOError: [Errno 4] Interrupted system call

```
</details>

--- a/test/common.py
+++ b/test/common.py
@@ -210,7 +210,7 @@ class TestCase(unittest.TestCase):
                         diff = diff.abs()
                     max_err = diff.max()
                     self.assertLessEqual(max_err, prec, message)
-            self.assertEqual(x.is_sparse, y.is_sparse, message)
+            super(TestCase, self).assertEqual(x.is_sparse, y.is_sparse, message)
             if x.is_sparse:
                 x = self.safeCoalesce(x)
                 y = self.safeCoalesce(y)

--- a/test/common.py
+++ b/test/common.py
@@ -8,6 +8,8 @@ import contextlib
 from functools import wraps
 from itertools import product
 from copy import deepcopy
+from numbers import Number
+
 import __main__
 import errno
 
@@ -197,7 +199,7 @@ class TestCase(unittest.TestCase):
 
         if torch.is_tensor(x) and torch.is_tensor(y):
             def assertTensorsEqual(a, b):
-                super(TestCase, self).assertEqual(a.size(), b.size())
+                super(TestCase, self).assertEqual(a.size(), b.size(), message)
                 if a.numel() > 0:
                     b = b.type_as(a)
                     b = b.cuda(device=a.get_device()) if a.is_cuda else b.cpu()
@@ -219,20 +221,25 @@ class TestCase(unittest.TestCase):
             else:
                 assertTensorsEqual(x, y)
         elif isinstance(x, string_classes) and isinstance(y, string_classes):
-            super(TestCase, self).assertEqual(x, y)
+            super(TestCase, self).assertEqual(x, y, message)
         elif type(x) == set and type(y) == set:
-            super(TestCase, self).assertEqual(x, y)
+            super(TestCase, self).assertEqual(x, y, message)
         elif is_iterable(x) and is_iterable(y):
-            super(TestCase, self).assertEqual(len(x), len(y))
+            super(TestCase, self).assertEqual(len(x), len(y), message)
             for x_, y_ in zip(x, y):
                 self.assertEqual(x_, y_, prec, message)
-        else:
-            try:
-                self.assertLessEqual(abs(x - y), prec, message)
+        elif isinstance(x, Number) and isinstance(y, Number):
+            if abs(x) == float('inf') and abs(y) == float('inf'):
                 return
-            except (TypeError, AssertionError):
-                pass
+            super(TestCase, self).assertLessEqual(abs(x - y), prec, message)
+        else:
             super(TestCase, self).assertEqual(x, y, message)
+
+    def assertAlmostEqual(self, x, y, places=None, msg=None, delta=None):
+        prec = delta
+        if places:
+            prec = 10**(-places)
+        self.assertEqual(x, y, prec, msg)
 
     def assertNotEqual(self, x, y, prec=None, message=''):
         if prec is None:

--- a/test/common.py
+++ b/test/common.py
@@ -229,9 +229,8 @@ class TestCase(unittest.TestCase):
             for x_, y_ in zip(x, y):
                 self.assertEqual(x_, y_, prec, message)
         elif isinstance(x, Number) and isinstance(y, Number):
-            if abs(x) == float('inf') and abs(y) == float('inf'):
-                return
-            super(TestCase, self).assertLessEqual(abs(x - y), prec, message)
+            if x != y:
+                super(TestCase, self).assertLessEqual(abs(x - y), prec, message)
         else:
             super(TestCase, self).assertEqual(x, y, message)
 

